### DISCUSSION TITLE
useIsSubmitting with formId triggering on multiple forms

### DISCUSSION THREAD

**Discussion by cill-i-am** [airjp73/rvf#234] [Q&A]
I have a parent route and a child modal route, both with forms with specified Ids using the useIsSubmitting(formId) hook to manage their loading state. When submitting the modal route the boolean is true for both forms though. I was under the impression that isSubmitting would only be true for the associated form Id, can anyone tell me where i'm going wrong?

```TS
const formId = 'required-documents-form';
const isSubmitting = useIsSubmitting(formId);

<ValidatedForm
   id={formId}
   method='post'
>
...
   <button
       form={formId}
       type='submit'
       className={
           `btn no-animation w-full md:w-48 
             ${isSubmitting ? 'loading disabled' : ''}
            `}
   >
      Finish
   </button>
</ValidatedForm>
....

```

https://user-images.githubusercontent.com/8224386/212548215-a765d666-c411-483b-95a8-87220341447c.mov

**Comment by airjp73** [comment]
> I was under the impression that isSubmitting would only be true for the associated form Id

That's correct. Is there any chance you're passing the same formId to both forms or that one of the buttons isn't being passed a formId at all? If not, could you provide a reproduction of the issue so we can fix it?

**Reply by cill-i-am** [reply]
Just double checked there and the underlying form/button/hook is receiving the correct formId. I'll make a repro after work and link an issue here for you. Thanks for the help!
